Automattic/jetpack
# Social Connections: Select the reconnecting account by default on confirmation screen

When reconnecting a broken connection, the confirmation screen does not auto select the appropriate account, it rather selects the first account.

## Proposed changes:

* Preselect the reconnecting account on the confirmation screen in connections management

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







<details>
    <summary>To simulate a broken connection, you can make these changes (click to expand)</summary>

```diff
diff --git a/projects/packages/publicize/src/class-publicize.php b/projects/packages/publicize/src/class-publicize.php
index be4070e853..d921f6a1da 100644
--- a/projects/packages/publicize/src/class-publicize.php
+++ b/projects/packages/publicize/src/class-publicize.php
@@ -255,7 +255,7 @@ class Publicize extends Publicize_Base {
                                                                        'can_disconnect' => self::can_manage_connection( $connection['connection_data'] ),
                                                                        'profile_link'   => $this->get_profile_link( $service_name, $connection ),
                                                                        'shared'         => $connection['connection_data']['user_id'] === '0',
-                                                                       'status'         => 'ok',
+                                                                       'status'         => 'broken',
                                                                )
                                                        );
                                                } else {
@@ -287,7 +287,7 @@ class Publicize extends Publicize_Base {
                $connection_results_map = array();

                foreach ( $connection_results as $connection_result ) {
-                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'ok' : 'broken';
+                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'broken' : 'broken';
                }
                foreach ( $connections as $key => $connection ) {
                        if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
```
</details>


* Go to connections management in Social, Jetpack settings and the editor
* Click on Reconnect for an account which ends up not the first account on the confirmation screen
* Confirm that the reconnecting account is preselected instead of the first one
* Add an account normally, without reconnecting
* Confirm that the first account is preselcted on the confirmation screen.


https://github.com/Automattic/jetpack/assets/18226415/4e0e9cd5-f182-4dbd-a045-702e62c18c39
